pkg/dapr/pubsub.go

```diff
@@ -0,0 +1,95 @@
+package dapr
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/dapr/go-sdk/client"
+	"github.com/dapr/go-sdk/service/common"
+	daprd "github.com/dapr/go-sdk/service/grpc"
+	"github.com/sirupsen/logrus"
+)
+
+type PubSub struct {
+	client      client.Client
+	Logger      *logrus.Logger
+	Timestamp   time.Time
+	Description string
+	Requests    int64
+}
+
+func InitPubsub(description string, pubsubName string, subPort string, pubPort string, logger *logrus.Logger, done chan struct{}) (*PubSub, error) {
+	var err error
+	clientPub, err := client.NewClientWithPort(pubPort)
+	if err != nil {
+		logger.Fatal(err)
+	}
+	pubSub := PubSub{
+		Logger:      logger,
+		Description: description,
+		client:      clientPub,
+		Timestamp:   time.Now(),
+		Requests:    0,
+	}
+	s, err := daprd.NewService(subPort)
+	if err != nil {
+		log.Fatalf("failed to start the server: %v", err)
+	}
+	sub := &common.Subscription{
+		PubsubName: "messages",
+		Topic:      "info",
```

Comment:
I assume, these values should be configurable (see config.go for reference)

Reply:
Done